core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -267,25 +249,30 @@ public GenericContainer() {
     }
 
     public GenericContainer(@NonNull final String dockerImageName) {
-        this.setDockerImageName(dockerImageName);
+        this(new RemoteDockerImage(DockerImageName.parse(dockerImageName)));
     }
 
     public GenericContainer(@NonNull final Future<String> image) {
-        setImage(image);
+        this(new RemoteDockerImage(image));
+    }
+
+    public GenericContainer(@NonNull final ContainerDef containerDef) {
```

Comment:
`ContainerDef` is internal. Also, we probably will have a different mechanism to instantiate `GenericContainer` from `ContainerDef`. Let's maybe remove this constructor, or make internal?